metron-analytics/metron-profiler-client/src/test/java/org/apache/metron/profiler/client/stellar/GetProfileTest.java

```diff
@@ -20,437 +20,182 @@
-    state = new HashMap<>();
-    final HTableInterface table = MockHBaseTableProvider.addToCache(tableName, columnFamily);
-
-    // used to write values to be read during testing
-    long periodDurationMillis = TimeUnit.MINUTES.toMillis(15);
-    RowKeyBuilder rowKeyBuilder = new SaltyRowKeyBuilder();
-    ColumnBuilder columnBuilder = new ValueOnlyColumnBuilder(columnFamily);
-    profileWriter = new ProfileWriter(rowKeyBuilder, columnBuilder, table, periodDurationMillis);
-
-    // global properties
-    Map<String, Object> global = new HashMap<String, Object>() {{
-      put(PROFILER_HBASE_TABLE.getKey(), tableName);
-      put(PROFILER_COLUMN_FAMILY.getKey(), columnFamily);
-      put(PROFILER_HBASE_TABLE_PROVIDER.getKey(), MockHBaseTableProvider.class.getName());
-      put(PROFILER_PERIOD.getKey(), Long.toString(periodDuration));
-      put(PROFILER_PERIOD_UNITS.getKey(), periodUnits.toString());
-      put(PROFILER_SALT_DIVISOR.getKey(), Integer.toString(saltDivisor));
-    }};
+    // the mock profiler client used to feed profile measurement values to the function
+    profilerClient = mock(ProfilerClient.class);
 
-    // create the stellar execution environment
-    executor = new DefaultStellarStatefulExecutor(
-            new SimpleFunctionResolver()
-                    .withClass(GetProfile.class)
-                    .withClass(FixedLookback.class),
-            new Context.Builder()
-                    .with(Context.Capabilities.GLOBAL_CONFIG, () -> global)
-                    .build());
-  }
-
-  /**
-   * This method is similar to setup(), in that it sets up profiler configuration context,
-   * but only for the client.  Additionally, it uses periodDuration2, periodUnits2
-   * and saltDivisor2, instead of periodDuration, periodUnits and saltDivisor respectively.
-   *
-   * This is used in the unit tests that test the config_overrides feature of PROFILE_GET.
-   * In these tests, the context from @Before setup() is used to write the data, then the global
-   * context is changed to context2 (from this method).  Each test validates that a default read
-   * using global context2 then gets no valid results (as expected), and that a read using
-   * original context values in the PROFILE_GET config_overrides argument gets all expected results.
-   *
-   * @return context2 - The profiler client configuration context created by this method.
-   *    The context2 values are also set in the configuration of the StellarStatefulExecutor
-   *    stored in the global variable 'executor'.  However, there is no API for querying the
-   *    context values from a StellarStatefulExecutor, so we output the context2 Context object itself,
-   *    for validation purposes (so that its values can be validated as being significantly
-   *    different from the setup() settings).
-   */
-  private Context setup2() {
-    state = new HashMap<>();
+    // the VERBOSE_PROFILE function that will be tested
+    function = new GetProfile();
```

Comment:
I just find this approach more clear without having to write a separate Builder class that would require more code.

Just to highlight what I mean by "more clear."  For example, I think this code in the tests is more clear...
```
function = new GetProfile();
function.withProfilerClientFactory(globals -> profilerClient);
```

Than this...
```
function = new GetProfile(globals -> profilerClient);
```

When I look at the use of a lambda as a constructor argument, I am left asking myself, "what the heck is a `globals -> profilerClient`"?  When you see the first code example, you immediately know that the lambda is a `ProfilerClientFactory`.

Now I could get the same clarity by creating a separate Builder class as you mentioned.  But I have to write more code to do a Builder; maybe just tens of lines, but it is more code.  What are the benefits of using a Builder here?  I'll do it if the benefit is clear. I can see the benefits of a Builder when there are lots of things that need plugged in, but here it is just once; a `ProfilerClientFactory`.

Reply:
That is my logic for taking this approach at least.  I could go either way. If you think the alternate constructor works, I'd be happy to compromise and knock this out.

Reply:
I will refactor this to a Builder.  No problem.

Reply:
This now uses the Builder Pattern.  See commit 14717703f.

Reply:
It's a good point you make about readability. At the very least, the variable names/args provide a mechanism for an IDE to give me either param details or javadoc, whereas object state requirements are less clear given any of the known IDE capabilities (save for extra documentation that might grow stale). I personally think emphasizing correct object state and encapsulation is a bit more important, though again I absolutely appreciate the rationale on readability and think that's very close behind. Appreciate the time adding the builder pattern impl. Looks like that wasn't too much to add, fortunately.

Reply:
Oops. I found a problem with how I did this.  Function initialization was failing in the REPL for `PROFILE_GET/VERBOSE`.  This was because Stellar function resolution uses the default constructor (and not the Builder.)  

When using the default constructor, the `profilerClientFactory` was not being initialized.  The tests did not fail because they were using the Builder, rather than the constructor directly.

Since we want the tests to function in as similar manner as production,  I think I should just use an alternate constructor instead of a Builder.  I don't want the tests using the Builder, but Stellar in production using the constructor.  I think this approach satisfies your concerns and actually works.

See 50309a140.